client/src/pages/CommunityPage/CommunityPageCreateGroup.js

```diff
@@ -0,0 +1,194 @@
+import Button from "../../components/Button/Button";
+import axios from "axios";
+import { BASE_URL } from "../../services/constants";
+import Autocomplete from "@material-ui/lab/Autocomplete";
+
+export default class CommunityPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: "",
+      description: "",
+      location: null,
+    };
+  }
+
+  handleChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
+  };
+
+  submit = () => {
+    console.log(this.state);
+    axios({
+      method: 'POST',
+      url: `${BASE_URL}/groups`,
+      data: {
+        title: this.state.title,
+        description: this.state.description,
+        location: this.state.location
+      }
+    })
+  };
+
+  componentDidMount() {
+    this.getUSStates();
+  }
+
+  getUSStates = () => {
+    axios.get(`${BASE_URL}/location/states`).then(res => {
+      console.log(res.data);
+      this.setState({ USStates: res.data });
+    });
+  };
+
+  getCitiesFromState = event => {
+    if (event.target.textContent === "") this.setState({ stateFound: false });
+
+    let USStates = this.state.USStates;
+
+    //TextContent because materialUI outputs an <li> tag
+    let userSubmission = event.target.textContent.toLowerCase();
+
+    USStates.forEach(state => {
+      if (state.name.toLowerCase() === userSubmission) {
+        axios.get(`${BASE_URL}/location/bystate?state__code=${state.code}`).then(res => {
+          this.setState({
+            stateFound: true,
+            currentStateCities: res.data
+          });
+        });
+      }
+    });
+  };
+
+  setLocationID = event => {
+    let currentStateCities = this.state.currentStateCities;
+    // console.log(currentStateCities);
+
+    let userSubmission = event.target.textContent.toLowerCase();
+
+    // for(let i=0;i<currentStateCities.length)
+
+    //binary search
+    let result = this.binarySearch(
+      currentStateCities,
+      userSubmission,
+      0,
+      currentStateCities.length - 1
+    );
+
+    if (result) this.setState({ location: result.location.id});
+    else this.setState({ location: null});
+
+    console.log("binarysearch: ", result);
+  };
+
+  binarySearch = (cities, userSubmission, start, end) => {
+    // Base Condition
+    if (start > end) return false;
+
+    // Find the middle index
+    let mid = Math.floor((start + end) / 2);
+
+    // Compare mid with given key x
+    if (cities[mid].name.toLowerCase() === userSubmission)
+      return { found: true, location: cities[mid] };
+
+    // If element at mid is greater than x,
+    // search in the left half of mid
+    if (cities[mid].name.toLowerCase() > userSubmission)
+      return this.binarySearch(cities, userSubmission, start, mid - 1);
+    // If element at mid is smaller than x,
+    // search in the right half of mid
+    else return this.binarySearch(cities, userSubmission, mid + 1, end);
+  };
```

Comment:
There has to be a better way to do this.

Reply:
This is the only part I have not had time to mess with yet, this also came from the account creation flow.